internal/controllers/clustertemplate_controller.go

```diff
@@ -96,31 +122,181 @@ func (r *ClusterTemplateReconciler) Reconcile(ctx context.Context, req ctrl.Requ
+	if validationErrsMsg != "" {
+		t.logger.ErrorContext(ctx, fmt.Sprintf(
+			"Failed to validate for ClusterTemplate %s: %s", t.object.Name, validationErrsMsg))
+	} else {
+		t.logger.InfoContext(ctx, fmt.Sprintf(
+			"Validation passing for ClusterTemplate %s", t.object.Name))
+	}
 
-	return utils.UpdateK8sCRStatus(ctx, t.client, t.object)
+	err = t.updateStatusConditionValidated(ctx, validationErrsMsg)
+	if err != nil {
+		return false, err
+	}
+	return validationErrsMsg == "", nil
+}
+
+// validateConfigmapReference validates a given configmap reference within the ClusterTemplate
+func validateConfigmapReference(
+	ctx context.Context, c client.Client, name, namespace, expectedKey string) (string, error) {
+
+	var validationErr string
+
+	existingConfigmap := &corev1.ConfigMap{}
+	cmExists, err := utils.DoesK8SResourceExist(
+		ctx, c, name, namespace, existingConfigmap)
+	if err != nil {
+		return validationErr, err
+	}
+
+	if !cmExists {
+		// Check if the configmap is missing
+		validationErr = fmt.Sprintf(
+			"The referenced ConfigMap %s is not found in the namespace %s", name, namespace)
+		return validationErr, nil
+	} else {
+		// Check if the expected key is present in the configmap data
+		defaults, exists := existingConfigmap.Data[utils.ClusterInstanceTemplateDefaultsConfigmapKey]
+		if !exists {
+			validationErr = fmt.Sprintf(
+				"the expected key %s does not exist in the ConfigMap %s data", expectedKey, name)
+			return validationErr, nil
+		}
+
+		// Check if the data value is valid YAML
+		var validData map[string]interface{}
+		err = yaml.Unmarshal([]byte(defaults), &validData)
+		if err != nil {
+			validationErr = fmt.Sprintf(
+				"the value of key %s from ConfigMap %s is not in a valid YAML string: %s", expectedKey, name, err.Error())
+			return validationErr, nil
+		}
+	}
+
+	// Check if the configmap is set to mutable
+	if existingConfigmap.Immutable != nil && !*existingConfigmap.Immutable {
+		validationErr = fmt.Sprintf(
+			"It is not allowed to set Immutable to false in the ConfigMap %s", name)
+		return validationErr, nil
+	} else if existingConfigmap.Immutable == nil {
+		// Patch the validated ConfigMap to make it immutable if not already set
+		immutable := true
+		newConfigmap := existingConfigmap.DeepCopy()
+		newConfigmap.Immutable = &immutable
+
+		if err = utils.CreateK8sCR(ctx, c, newConfigmap, nil, utils.PATCH); err != nil {
+			return validationErr, err
+		}
+	}
+
+	return validationErr, nil
+}
+
+// validateUniqueConfigmapReferences ensures that the Configmaps referenced in the ClusterTemplate are unique across all ClusterTemplates.
+// The required unique configmap - ClusterInstanceDefaults
+func (t *clusterTemplateReconcilerTask) validateUniqueConfigmapReferences(ctx context.Context) (string, error) {
+	var validationErr string
+	var ciDefaultsCmCts []string
+
+	clusterTemplates := &oranv1alpha1.ClusterTemplateList{}
+	err := t.client.List(ctx, clusterTemplates)
```

Comment:
I think here we should get the clusterTemplates in the same namespace rather than all the existing clusterTemplates in the cluster.